sphinx_immaterial/inline_icons.py

```diff
@@ -0,0 +1,66 @@
+"""This module adds the md-icon role. This is specific to this this theme because,
+primarily, the icons bundled with this theme are supported (plus any svg files in the
+user project's static directories)."""
+from pathlib import Path
+from typing import Tuple, List
+from docutils import nodes
+from sphinx.application import Sphinx
+from sphinx.builders import Builder
+from sphinx.writers.html5 import HTML5Translator
+
+
+def read_svg_into_span(builder: Builder, icon_name: str, classes: List[str]) -> str:
+    """Reads the SVG data into a HTML span element."""
+    svg = Path(__file__).parent / ".icons" / icon_name
+    if not svg.exists():
+        static_paths: List[str] = getattr(builder.config, "html_static_path")
+        for path in static_paths:
+            svg = Path(builder.srcdir) / path / icon_name
+            if svg.exists():
+                break
+        else:
+            raise FileNotFoundError(
+                f"{icon_name} not found in html_static_path and not bundled with theme"
+            )
+    svg_data = svg.read_text(encoding="utf-8")
+    css_classes = " ".join(classes)
+    return f'<span class="{css_classes}">{svg_data}</span>'
+
+
+class si_icon(nodes.container):
+    pass
+
+
+def visit_si_icon(self: HTML5Translator, node: si_icon):
+    """read icon data and put it into a span element"""
+    # in case this node was created from another extension
+    assert node.rawsource, "icon node's rawsource must be a relative path"
+
+    self.body.append(
+        read_svg_into_span(
+            builder=self.builder,
+            icon_name=node.rawsource + ".svg",
+            classes=node["classes"],
+        )
+    )
+    raise nodes.SkipNode()
+
+
+def icons_role(
+    role: str, rawtext: str, text: str, lineno: int, inliner, options={}, content=[]
+) -> Tuple[List[nodes.Node], List[nodes.system_message]]:
+    path, classes = (text, "")
+    if ";" in text:
```

Comment:
Are there other properties other than `class` that we may wish to support in the future?  If so it could make sense to use a syntax like `class=`.  On the other hand I guess we could just use the `name=value` syntax for those other options and keep the shorthand for `class`.

Reply:
This implementation is mostly based on sphinx-design inline icons. They have some other options about sizing (by pixels), but it is flawed because not all SVG data behave well with specific sizes and it requires a certain order of properties (`path;classes;sizes;color` or something like that).

I personally think that any other properties can be managed in CSS more efficiently, but the embedded data in a span is about to change. We could re-assess when this feature is more finalized with #183. I will add a slice notation to make sure that only 2 list items are expanded.

Reply:
One property that might be useful and cannot be handled from CSS is `title`, or perhaps `alt` if it is supported.

I imagine these icons would most commonly be used via a `|substitution|` and the title text could indicate the meaning of the icon, if the icon is intended to provide additional information rather than being purely decorative.

Anyway this is fine as is.